x/stakeibc/keeper/icacallbacks_undelegate.go

```diff
@@ -66,12 +64,12 @@ func (k Keeper) UndelegateCallback(ctx sdk.Context, packet channeltypes.Packet,
 		k.Logger(ctx).Error(utils.LogICACallbackStatusWithHostZone(chainId, ICACallbackID_Undelegate,
 			icacallbackstypes.AckResponseStatus_FAILURE, packet))
 
-		// Reset unbondings record status
+		// Set the unbonding status to RETRY
 		if err := k.RecordsKeeper.SetHostZoneUnbondingStatus(
 			ctx,
 			chainId,
 			undelegateCallback.EpochUnbondingRecordIds,
-			recordstypes.HostZoneUnbonding_UNBONDING_QUEUE,
+			recordstypes.HostZoneUnbonding_UNBONDING_RETRY_QUEUE,
```

Comment:
Checking my understanding: In the case of this unbonding record going back into the retry queue there is no need to re-marshal and save the batch data because those particular split Undelegations and batching only mattered for this particular ICA submission.  When this record gets retried the whole validator assignment and batching flow will happen again from scratch and new splitUndelegations will be made for that future batching correct?

Reply:
yup exactly - the callback data from this batch will be deleted as soon as the ack completes. And then new callback data will be created in the retry